Core/Security/m2crypto/X509Chain.py

```diff
@@ -1050,20 +1053,20 @@ def hash(self):
 
         :returns: S_OK(string hash)
     """
-
     if self.__hash:
       return S_OK(self.__hash)
     sha1 = hashlib.sha1()
     for cert in self._certList:
-      sha1.update(str(cert.getSubjectNameObject()))
```

Comment:
@chaen Currently this line is broken as `cert.getSubjectNameObject()` returns a `S_OK` object, i.e.

```python
(Pdb) str(cert.getSubjectNameObject()["Value"])
'/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=cburr/CN=761704/CN=Chris Burr/CN=5399599166/CN=2787290085/CN=8974817748'
(Pdb) str(cert.getSubjectNameObject())
"{'OK': True, 'Value': <M2Crypto.X509.X509_Name object at 0x7f24f0461ca0>}"
```

Should it be fixed in `rel-v7r0`?

Reply:
yeah I saw that. It can be but it is not strictly necessary. As per comment, this is used exclusively to generate a hash to store in a dict. So as long as it is deterministic and unique, it is fine. (yes, I am sure that there would have been much better way of doing :-D )

Reply:
I doubt the pointer (`0x7f24f0461ca0`) is deterministic?

Reply:
Good point, I overlooked the pointer :-) Then yes, please backport it :-)